content/exploratory_notebooks/facebook_notebook.md

```diff
@@ -0,0 +1,463 @@
+```
+
+Let's check the distribution of the eigenvector centralities:
+
+```{code-cell} ipython3
+plt.figure(figsize=(15,8))
+plt.hist(eigenvector_centrality.values(), bins=60)
+plt.xticks(ticks=[0, 0.01, 0.02, 0.04, 0.06, 0.08])  # set the x axis ticks
+plt.title('Eigenvector Centrality Histogram ', fontdict ={'size': 35}, loc='center') 
+plt.xlabel('Eigenvector Centrality', fontdict ={'size': 20})
+plt.ylabel('Counts',fontdict ={'size': 20})
+```
+
+As shown in the distribution histogram, the vast majority of eigenvector centralities are below $0.005$ and are actually almost $0$. However, we can also see different values of eigenvector centralities as there are tiny bins all over the x axis.
+
+Now we can identify the eigenvector centralities of nodes based on their size in the following representation:
+
+```{code-cell} ipython3
+node_size =  [v * 4000 for v in eigenvector_centrality.values()]  # set up nodes size for a nice graph representation
+plt.figure(figsize=(15,8))
+nx.draw_networkx(G, node_size=node_size, with_labels=False, width=0.15)
+plt.axis('off')
+```
+
+## Clustering Effects
+The clustering coefficient of a node $v$ is defined as the probability that two randomly selected friends of $v$ are friends with each other. As a result, the average clustering coefficient is the average of clustering coefficients of all the nodes. The closer the average clustering coefficient is to $1$, the more complete the graph will be because there's just one giant component. Lastly, it is a sign of triadic closure because the more complete the graph is, the more triangles will usually arise.
+
+```{code-cell} ipython3
+nx.average_clustering(G)
+```
+
+Now the clustering coefficient distribution will be displayed:
+
+```{code-cell} ipython3
+plt.figure(figsize=(15,8))
+plt.hist(nx.clustering(G).values(), bins=50)
+plt.title('Clustering Coefficient Histogram ', fontdict ={'size': 35}, loc='center') 
+plt.xlabel('Clustering Coefficient', fontdict ={'size': 20})
+plt.ylabel('Counts',fontdict ={'size': 20})
+```
+
+ $50$ bins were used to showcase the distribution. The bin with the highest counts concerns nodes with clustering coefficient close to $1$ as there are more than two-hundred-fifty nodes in that bin. In addition, the bins of clustering coefficient between $0.4$ and $0.8$ contain the majority of nodes by far. 
+ 
+ The number of unique triangles in the network are found next:
+
+```{code-cell} ipython3
+sum(list(nx.triangles(G).values())) / 3  # divide by 3 because each triangle is counted once for each node
+```
+
+ Now the average number of triangles that a node is a part of:
+
+```{code-cell} ipython3
+np.mean(list(nx.triangles(G).values()))
+```
+
+Due to having some nodes that belong to a great many triangles, the metric of median will give us a better understanding:
+
+```{code-cell} ipython3
+np.median(list(nx.triangles(G).values()))
+```
```

Comment:
````suggestion
```{code-cell} ipython3
triangles_per_node = list(nx.triangles(G).values())
sum(triangles_per_node) / 3  # divide by 3 because each triangle is counted once for each node
```

 Now the average number of triangles that a node is a part of:

```{code-cell} ipython3
np.mean(triangles_per_node)
```

Due to having some nodes that belong to a great many triangles, the metric of median will give us a better understanding:

```{code-cell} ipython3
np.median(triangles_per_node)
```
````
We can get a minor performance boost by reusing the result of `nx.triangles` rather than recomputing it each time.